src/Streamer.cpp

```diff
@@ -199,76 +197,88 @@ void Streamer::processMessage(
     // are already marked as reached
     if (stopOffsetsReached()) {
       Logger->warn("There was no data in {} to consume", ConsumerTopicName);
-      RunStatus.store(StreamerStatus::HAS_FINISHED);
-      return;
+      return true;
     }
   }
 
-  if (KafkaMessage->first == KafkaW::PollStatus::Error) {
-    // TODO count Kafka error
-    return;
+  if (CatchingUpToStopOffset && stopOffsetsNowReached(Partition, Offset)) {
+    Logger->debug("Reached stop offsets in topic {}", ConsumerTopicName);
+    return true;
   }
 
-  if (KafkaMessage->first == KafkaW::PollStatus::Empty ||
-      KafkaMessage->first == KafkaW::PollStatus::EndOfPartition ||
-      KafkaMessage->first == KafkaW::PollStatus::TimedOut) {
-    return;
+  return false;
+}
+
+bool Streamer::kafkaMessageIsValid(KafkaW::PollStatus MessageStatus) {
```

Comment:
I'm not too keen on the name of this function because it sounds to me like we are checking the contents of the message payload. Really we are checking whether an actual message payload has been returned at all. Maybe `messageHasPayload()`?

Reply:
👍 Makes sense - thanks